h2/src/test/org/h2/test/jdbc/TestPreparedStatement.java

```diff
@@ -1403,9 +1403,9 @@ private void testObject(Connection conn) throws SQLException {
         assertTrue(rs.getObject(2).equals("Abc"));
         assertTrue(rs.getObject(3).equals(new BigDecimal("10.2")));
         assertTrue(rs.getObject(4).equals(SysProperties.OLD_RESULT_SET_GET_OBJECT ?
-                (Object) Byte.valueOf((byte) 0xff) : (Object) Integer.valueOf(-1)));
+                (byte) 0xff : -1));
```

Comment:
Please, run this test case with `java -Dh2.oldResultSetGetObject=false org.h2.test.jdbc.TestPreparedStatement`.

You broke the test case for this flag. All your changes in this test case are invalid, they do not produce the same bytecode as before.